### DISCUSSION TITLE
Letting Vim change the window size

### DISCUSSION THREAD

**Discussion by chdiza** [kovidgoyal/kitty#4206] [Q&A] [answered]
In `uxterm`, I can change the size of the terminal window "automatically" when I set `&rows` or `&columns` inside of Vim.  I've had mappings since my macOS days that do things like `:set columns=75 lines=24` in Vim.  If the terminal window is not  75 columns or not  24 rows, then the Vim window *and* the terminal window both get made into 75x24.   I got accustomed to this on macOS, where Terminal.app and iTerm2 both allow this.  Now that I've switched to linux, it appears that `uxterm` allows this but kitty does not.

Is there a kitty setting that can allow the program running inside kitty to change the window size?

**Comment by kovidgoyal** [comment, marked as answer]
No there isnt. Windows cannot generally be resized freely. They are
amanged by layouts and kitty OS Windows themeselves may not be resizable
if you use a tiling window manager.

On Sun, Nov 07, 2021 at 11:04:51AM -0800, chdiza wrote:
> In `uxterm`, I can change the size of the terminal window "automatically" when I set `&rows` or `&columns` inside of Vim.  I've had mappings since my macOS days that do things like `:set columns=75 lines=24` in Vim.  If the terminal window is not  75 columns or not  24 rows, then the Vim window *and* the terminal window both get made into 75x24.   I got accustomed to this on macOS, where Terminal.app and iTerm2 both allow this.  Now that I've switched to linux, it appears that `uxterm` allows this but kitty does not.
> 
> Is there a kitty setting that can allow the program running inside kitty to change the window size?
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/4206

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by chdiza** [comment]
OK, thanks for the answer.

In awesome-wm, I can freely resize kitty's "os windows".  So I'll just use that instead of using Vim mappings; they both take me to the same place in the end.
